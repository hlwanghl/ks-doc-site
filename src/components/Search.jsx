import React from 'react'
import styled from 'styled-components'

import { ReactComponent as SearchIcon } from '../assets/search.svg'

class Search extends React.Component {

  handleKeyUp = (e) => {
    if (e.keyCode === 13) {
      const query = e.target.value
      this.props.onSearch(query)
    }
  }

  handleChange = (e) => {
    this.props.onQueryChange(e.target.value)
  }

  render() {
    return (
      <SearchWrapper className="ks-search">
        <SearchIcon />
        <input 
          type="text" 
          value={this.props.query} 
          placeholder="快速查找" 
          onKeyUp={this.handleKeyUp}
          onChange={this.handleChange}
        />
      </SearchWrapper>
    );
  }
}

const SearchWrapper = styled.div`
  position: absolute;
  right: 20px;
  top: 50%;
  transform: translateY(-50%);
  width: 350px;
  height: 36px;
  

  .algolia-autocomplete {
    width: 100%;
    height: 100%;
    background-color: transparent;
  }

  .algolia-autocomplete .ds-dropdown-menu {
    transform: translateY(6px);
  }

  input {
    width: 100%;
    height: 100%;
    padding: 7px 20px 7px 40px;
    font-size: 0.875rem;
    font-family: Proxima Nova;
    font-weight: 600;
    line-height: 1.7;
    color: #303e5a;
    border-radius: 18px;
    border: solid 1px #cfd9df;
    background-color: transparent;
    transition: all .2s ease;

    &::placeholder {
      color: #657d95;
      font-weight: normal;
    }

    &:focus {
      outline: none;
      background-color: #fff;
    }
  }

  & > svg {
    position: absolute;
    top: 50%;
    left: 14px;
    width: 14px;
    height: 14px;
    transform: translateY(-50%);
    z-index: 2;
  }
`

export default Search
