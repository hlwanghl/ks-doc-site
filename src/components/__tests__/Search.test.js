import React from 'react'
import renderer from 'react-test-renderer'
import Search from '../Search'

describe('Search', () => {
  it('renders correctly', () => {
    const props = {}

    const tree = renderer.create(<Search {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
