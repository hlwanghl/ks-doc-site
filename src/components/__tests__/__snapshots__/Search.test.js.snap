// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="ks-search sc-bdVaJa ewAAcf"
>
  <SvgMock />
  <input
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="快速查找"
    type="text"
  />
</div>
`;
